apps/openmw/mwmechanics/spellcasting.cpp

```diff
@@ -763,6 +763,35 @@ namespace MWMechanics
 
         int school = 0;
 
+        if(mCaster.getClass().isActor()
+                && MWBase::Environment::get().getWorld()->getStore().get<ESM::MagicEffect>()
+                .find(ESM::MagicEffect::WaterWalking))
```

Comment:
I'm not sure what you are trying to do here. The WaterWalking effect will always exist in the Store (with valid game files, anyway), so there's no point in checking it.